### ISSUE TITLE
[bug] Missing requirement to build DonPAPI through poetry: ModuleNotFoundError: No module named 'swig'

### ISSUE THREAD

**Issue by p0dalirius** [login-securite/DonPAPI#61]
Missing requirement 'swig' to build DonPAPI through poetry

```
┌──(root㉿kali)-[/home/kali/Desktop/DonPAPI]
└─# poetry update
Creating virtualenv donpapi-WjBzr8W3-py3.11 in /root/.cache/pypoetry/virtualenvs
Updating dependencies
Resolving dependencies... Downloading https://files.pythonhosted.org/packages/15/b8/d99814bab6797ed0b9449390c8c2f9872d475c693d11957b8b61abbc0933/impacket-0.9.24.tar.gz  18% (6.Resolving dependencies... Downloading https://files.pythonhosted.org/packages/15/b8/d99814bab6797ed0b9449390c8c2f9872d475c693d11957b8b61abbc0933/impacket-0.9.24.tar.gz  95% (6.Resolving dependencies... Downloading https://files.pythonhosted.org/packages/a1/eb/030d2f273902f2b0c7d59bca28fe528dcc7213f2ddd5a65055a3f25101be/pycryptodome-3.18.0-cp27-cp27m-Resolving dependencies... Downloading https://files.pythonhosted.org/packages/7c/6f/fdf4620511f30482915c88cba42e82b2ce3444499bfeba572c16abfd22e6/pycryptodomex-3.18.0-cp27-cp27mResolving dependencies... (13.1s)

Writing lock file

Package operations: 31 installs, 1 update, 0 removals

  • Installing idna (3.4)
  • Installing sniffio (1.3.0)
  • Installing anyio (3.7.1)
  • Installing certifi (2023.7.22)
  • Installing h11 (0.14.0)
  • Installing pycparser (2.21)
  • Installing cffi (1.15.1)
  • Installing httpcore (0.17.3)
  • Installing markupsafe (2.1.3)
  • Installing pyasn1 (0.5.0)
  • Installing blinker (1.6.2)
  • Installing click (8.1.6)
  • Installing cryptography (41.0.2)
  • Installing dnspython (2.4.0)
  • Installing future (0.18.3)
  • Installing itsdangerous (2.1.2)
  • Installing jinja2 (3.1.2)
  • Installing ldap3 (2.9.1)
  • Installing werkzeug (2.3.6)
  • Installing chardet (5.1.0)
  • Installing flask (2.3.2)
  • Installing six (1.16.0)
  • Installing pycryptodomex (3.18.0)
  • Installing pyopenssl (23.2.0)
  • Installing ldapdomaindump (0.9.4)
  • Installing impacket (0.9.24): Installing...
  • Installing lnkparse3 (1.2.0)
  • Installing m2crypto (0.38.0): Failed

  CalledProcessError

  Command '['/root/.cache/pypoetry/virtualenvs/donpapi-WjBzr8W3-py3.11/bin/python', '-m', 'pip', 'install', '--use-pep517', '--disable-pip-version-check', '--isolated', '--no-input', '--prefix', '/root/.cache/pypoetry/virtualenvs/donpapi-WjBzr8W3-py3.11', '--no-deps', '/root/.cache/pypoetry/artifacts/de/e9/1c/53d251c759c702dbaaaaca584d20259e0dac1d89660e8cf5208e6d1183/M2Crypto-0.38.0.tar.gz']' returned non-zero exit status 1.

  at /usr/lib/python3.11/subprocess.py:571 in run
       567│             # We don't call process.wait() as .__exit__ does that for us.
       568│             raise
       569│         retcode = process.poll()
       570│         if check and retcode:
    →  571│             raise CalledProcessError(retcode, process.args,
       572│                                      output=stdout, stderr=stderr)
       573│     return CompletedProcess(process.args, retcode, stdout, stderr)
       574│ 
       575│ 

The following error occurred when trying to handle this error:


  EnvCommandError

  Command ['/root/.cache/pypoetry/virtualenvs/donpapi-WjBzr8W3-py3.11/bin/python', '-m', 'pip', 'install', '--use-pep517', '--disable-pip-version-check', '--isolated', '--no-input', '--prefix', '/root/.cache/pypoetry/virtualenvs/donpapi-WjBzr8W3-py3.11', '--no-deps', '/root/.cache/pypoetry/artifacts/de/e9/1c/53d251c759c702dbaaaaca584d20259e0dac1d89660e8cf5208e6d1183/M2Crypto-0.38.0.tar.gz'] errored with the following return code 1, and output: 
  Processing /root/.cache/pypoetry/artifacts/de/e9/1c/53d251c759c702dbaaaaca584d20259e0dac1d89660e8cf5208e6d1183/M2Crypto-0.38.0.tar.gz
    Installing build dependencies: started
    Installing build dependencies: finished with status 'done'
    Getting requirements to build wheel: started
    Getting requirements to build wheel: finished with status 'done'
    Preparing metadata (pyproject.toml): started
    Preparing metadata (pyproject.toml): finished with status 'done'
  Building wheels for collected packages: M2Crypto
    Building wheel for M2Crypto (pyproject.toml): started
    Building wheel for M2Crypto (pyproject.toml): finished with status 'error'
    error: subprocess-exited-with-error
    
    × Building wheel for M2Crypto (pyproject.toml) did not run successfully.
    │ exit code: 1
    ╰─> [64 lines of output]
        INFO:run_command:running bdist_wheel
        INFO:run_command:running build
        INFO:run_command:running build_py
        INFO:copy_file:copying src/M2Crypto/DH.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/threading.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/X509.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/EC.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/AuthCookie.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/six.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/m2crypto.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/BIO.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/DSA.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/m2urllib.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/m2urllib2.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/m2xmlrpclib.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/SMIME.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/ASN1.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/__init__.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/m2.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/BN.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/RC4.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/Rand.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/callback.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/util.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/Err.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/Engine.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/EVP.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/ftpslib.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/RSA.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/httpslib.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:mkpath:creating build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/SSLServer.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/Checker.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/Session.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/Cipher.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/Connection.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/timeout.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/TwistedProtocolWrapper.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/Context.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/__init__.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/ssl_dispatcher.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/cb.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:run_command:running egg_info
        INFO:write_pkg_info:writing src/M2Crypto.egg-info/PKG-INFO
        INFO:write_file:writing dependency_links to src/M2Crypto.egg-info/dependency_links.txt
        INFO:write_file:writing top-level names to src/M2Crypto.egg-info/top_level.txt
        INFO:read_manifest:reading manifest file 'src/M2Crypto.egg-info/SOURCES.txt'
        INFO:read_template:reading manifest template 'MANIFEST.in'
        WARNING:process_template_line:warning: no files found matching 'SWIG/*.i'
        WARNING:process_template_line:warning: no files found matching 'SWIG/*.h'
        WARNING:process_template_line:warning: no files found matching 'SWIG/*.def'
        WARNING:process_template_line:warning: no files found matching 'SWIG/_m2crypto_wrap.c'
        WARNING:process_template_line:warning: no files found matching 'M2Crypto/m2crypto.py'
        INFO:add_license_files:adding license file 'LICENCE'
        INFO:execute:writing manifest file 'src/M2Crypto.egg-info/SOURCES.txt'
        INFO:run_command:running build_ext
        INFO:build_extension:building 'M2Crypto._m2crypto' extension
        INFO:swig_sources:swigging src/SWIG/_m2crypto.i to src/SWIG/_m2crypto_wrap.c
        INFO:spawn:swig -python -py3 -I/usr/lib/gcc/x86_64-linux-gnu/12/include -I/usr/local/include -I/usr/include/x86_64-linux-gnu -I/usr/include -D__x86_64__ -I/root/.cache/pypoetry/virtualenvs/donpapi-WjBzr8W3-py3.11/include -I/usr/include/python3.11 -I/usr/include/openssl -includeall -modern -builtin -outdir /tmp/pip-req-build-7ji35ain/src/M2Crypto -o src/SWIG/_m2crypto_wrap.c src/SWIG/_m2crypto.i
        Traceback (most recent call last):
          File "/root/.cache/pypoetry/virtualenvs/donpapi-WjBzr8W3-py3.11/bin/swig", line 5, in <module>
            from swig import swig
        ModuleNotFoundError: No module named 'swig'
        error: command '/root/.cache/pypoetry/virtualenvs/donpapi-WjBzr8W3-py3.11/bin/swig' failed with exit code 1
  • Installing lnkparse3 (1.2.0)
  • Installing m2crypto (0.38.0): Failed

  CalledProcessError

  Command '['/root/.cache/pypoetry/virtualenvs/donpapi-WjBzr8W3-py3.11/bin/python', '-m', 'pip', 'install', '--use-pep517', '--disable-pip-version-check', '--isolated', '--no-input', '--prefix', '/root/.cache/pypoetry/virtualenvs/donpapi-WjBzr8W3-py3.11', '--no-deps', '/root/.cache/pypoetry/artifacts/de/e9/1c/53d251c759c702dbaaaaca584d20259e0dac1d89660e8cf5208e6d1183/M2Crypto-0.38.0.tar.gz']' returned non-zero exit status 1.

  at /usr/lib/python3.11/subprocess.py:571 in run
       567│             # We don't call process.wait() as .__exit__ does that for us.
       568│             raise
       569│         retcode = process.poll()
       570│         if check and retcode:
    →  571│             raise CalledProcessError(retcode, process.args,
       572│                                      output=stdout, stderr=stderr)
       573│     return CompletedProcess(process.args, retcode, stdout, stderr)
       574│ 
       575│ 

The following error occurred when trying to handle this error:


  EnvCommandError

  Command ['/root/.cache/pypoetry/virtualenvs/donpapi-WjBzr8W3-py3.11/bin/python', '-m', 'pip', 'install', '--use-pep517', '--disable-pip-version-check', '--isolated', '--no-input', '--prefix', '/root/.cache/pypoetry/virtualenvs/donpapi-WjBzr8W3-py3.11', '--no-deps', '/root/.cache/pypoetry/artifacts/de/e9/1c/53d251c759c702dbaaaaca584d20259e0dac1d89660e8cf5208e6d1183/M2Crypto-0.38.0.tar.gz'] errored with the following return code 1, and output: 
  Processing /root/.cache/pypoetry/artifacts/de/e9/1c/53d251c759c702dbaaaaca584d20259e0dac1d89660e8cf5208e6d1183/M2Crypto-0.38.0.tar.gz
    Installing build dependencies: started
    Installing build dependencies: finished with status 'done'
    Getting requirements to build wheel: started
    Getting requirements to build wheel: finished with status 'done'
    Preparing metadata (pyproject.toml): started
    Preparing metadata (pyproject.toml): finished with status 'done'
  Building wheels for collected packages: M2Crypto
    Building wheel for M2Crypto (pyproject.toml): started
    Building wheel for M2Crypto (pyproject.toml): finished with status 'error'
    error: subprocess-exited-with-error
    
    × Building wheel for M2Crypto (pyproject.toml) did not run successfully.
    │ exit code: 1
    ╰─> [64 lines of output]
        INFO:run_command:running bdist_wheel
        INFO:run_command:running build
        INFO:run_command:running build_py
        INFO:copy_file:copying src/M2Crypto/DH.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/threading.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/X509.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/EC.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/AuthCookie.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/six.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/m2crypto.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/BIO.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/DSA.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/m2urllib.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/m2urllib2.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/m2xmlrpclib.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/SMIME.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/ASN1.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/__init__.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/m2.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/BN.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/RC4.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/Rand.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/callback.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/util.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/Err.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/Engine.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/EVP.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/ftpslib.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/RSA.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/httpslib.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:mkpath:creating build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/SSLServer.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/Checker.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/Session.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/Cipher.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/Connection.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/timeout.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/TwistedProtocolWrapper.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/Context.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/__init__.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/ssl_dispatcher.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/cb.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:run_command:running egg_info
        INFO:write_pkg_info:writing src/M2Crypto.egg-info/PKG-INFO
        INFO:write_file:writing dependency_links to src/M2Crypto.egg-info/dependency_links.txt
        INFO:write_file:writing top-level names to src/M2Crypto.egg-info/top_level.txt
        INFO:read_manifest:reading manifest file 'src/M2Crypto.egg-info/SOURCES.txt'
        INFO:read_template:reading manifest template 'MANIFEST.in'
        WARNING:process_template_line:warning: no files found matching 'SWIG/*.i'
        WARNING:process_template_line:warning: no files found matching 'SWIG/*.h'
        WARNING:process_template_line:warning: no files found matching 'SWIG/*.def'
        WARNING:process_template_line:warning: no files found matching 'SWIG/_m2crypto_wrap.c'
        WARNING:process_template_line:warning: no files found matching 'M2Crypto/m2crypto.py'
        INFO:add_license_files:adding license file 'LICENCE'
        INFO:execute:writing manifest file 'src/M2Crypto.egg-info/SOURCES.txt'
        INFO:run_command:running build_ext
        INFO:build_extension:building 'M2Crypto._m2crypto' extension
        INFO:swig_sources:swigging src/SWIG/_m2crypto.i to src/SWIG/_m2crypto_wrap.c
        INFO:spawn:swig -python -py3 -I/usr/lib/gcc/x86_64-linux-gnu/12/include -I/usr/local/include -I/usr/include/x86_64-linux-gnu -I/usr/include -D__x86_64__ -I/root/.cache/pypoetry/virtualenvs/donpapi-WjBzr8W3-py3.11/include -I/usr/include/python3.11 -I/usr/include/openssl -includeall -modern -builtin -outdir /tmp/pip-req-build-7ji35ain/src/M2Crypto -o src/SWIG/_m2crypto_wrap.c src/SWIG/_m2crypto.i
        Traceback (most recent call last):
          File "/root/.cache/pypoetry/virtualenvs/donpapi-WjBzr8W3-py3.11/bin/swig", line 5, in <module>
            from swig import swig
        ModuleNotFoundError: No module named 'swig'
        error: command '/root/.cache/pypoetry/virtualenvs/donpapi-WjBzr8W3-py3.11/bin/swig' failed with exit code 1
  • Installing impacket (0.9.24)
  • Installing lnkparse3 (1.2.0)
  • Installing m2crypto (0.38.0): Failed

  CalledProcessError

  Command '['/root/.cache/pypoetry/virtualenvs/donpapi-WjBzr8W3-py3.11/bin/python', '-m', 'pip', 'install', '--use-pep517', '--disable-pip-version-check', '--isolated', '--no-input', '--prefix', '/root/.cache/pypoetry/virtualenvs/donpapi-WjBzr8W3-py3.11', '--no-deps', '/root/.cache/pypoetry/artifacts/de/e9/1c/53d251c759c702dbaaaaca584d20259e0dac1d89660e8cf5208e6d1183/M2Crypto-0.38.0.tar.gz']' returned non-zero exit status 1.

  at /usr/lib/python3.11/subprocess.py:571 in run
       567│             # We don't call process.wait() as .__exit__ does that for us.
       568│             raise
       569│         retcode = process.poll()
       570│         if check and retcode:
    →  571│             raise CalledProcessError(retcode, process.args,
       572│                                      output=stdout, stderr=stderr)
       573│     return CompletedProcess(process.args, retcode, stdout, stderr)
       574│ 
       575│ 

The following error occurred when trying to handle this error:


  EnvCommandError

  Command ['/root/.cache/pypoetry/virtualenvs/donpapi-WjBzr8W3-py3.11/bin/python', '-m', 'pip', 'install', '--use-pep517', '--disable-pip-version-check', '--isolated', '--no-input', '--prefix', '/root/.cache/pypoetry/virtualenvs/donpapi-WjBzr8W3-py3.11', '--no-deps', '/root/.cache/pypoetry/artifacts/de/e9/1c/53d251c759c702dbaaaaca584d20259e0dac1d89660e8cf5208e6d1183/M2Crypto-0.38.0.tar.gz'] errored with the following return code 1, and output: 
  Processing /root/.cache/pypoetry/artifacts/de/e9/1c/53d251c759c702dbaaaaca584d20259e0dac1d89660e8cf5208e6d1183/M2Crypto-0.38.0.tar.gz
    Installing build dependencies: started
    Installing build dependencies: finished with status 'done'
    Getting requirements to build wheel: started
    Getting requirements to build wheel: finished with status 'done'
    Preparing metadata (pyproject.toml): started
    Preparing metadata (pyproject.toml): finished with status 'done'
  Building wheels for collected packages: M2Crypto
    Building wheel for M2Crypto (pyproject.toml): started
    Building wheel for M2Crypto (pyproject.toml): finished with status 'error'
    error: subprocess-exited-with-error
    
    × Building wheel for M2Crypto (pyproject.toml) did not run successfully.
    │ exit code: 1
    ╰─> [64 lines of output]
        INFO:run_command:running bdist_wheel
        INFO:run_command:running build
        INFO:run_command:running build_py
        INFO:copy_file:copying src/M2Crypto/DH.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/threading.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/X509.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/EC.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/AuthCookie.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/six.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/m2crypto.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/BIO.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/DSA.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/m2urllib.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/m2urllib2.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/m2xmlrpclib.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/SMIME.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/ASN1.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/__init__.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/m2.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/BN.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/RC4.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/Rand.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/callback.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/util.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/Err.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/Engine.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/EVP.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/ftpslib.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/RSA.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:copy_file:copying src/M2Crypto/httpslib.py -> build/lib.linux-x86_64-cpython-311/M2Crypto
        INFO:mkpath:creating build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/SSLServer.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/Checker.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/Session.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/Cipher.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/Connection.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/timeout.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/TwistedProtocolWrapper.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/Context.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/__init__.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/ssl_dispatcher.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:copy_file:copying src/M2Crypto/SSL/cb.py -> build/lib.linux-x86_64-cpython-311/M2Crypto/SSL
        INFO:run_command:running egg_info
        INFO:write_pkg_info:writing src/M2Crypto.egg-info/PKG-INFO
        INFO:write_file:writing dependency_links to src/M2Crypto.egg-info/dependency_links.txt
        INFO:write_file:writing top-level names to src/M2Crypto.egg-info/top_level.txt
        INFO:read_manifest:reading manifest file 'src/M2Crypto.egg-info/SOURCES.txt'
        INFO:read_template:reading manifest template 'MANIFEST.in'
        WARNING:process_template_line:warning: no files found matching 'SWIG/*.i'
        WARNING:process_template_line:warning: no files found matching 'SWIG/*.h'
        WARNING:process_template_line:warning: no files found matching 'SWIG/*.def'
        WARNING:process_template_line:warning: no files found matching 'SWIG/_m2crypto_wrap.c'
        WARNING:process_template_line:warning: no files found matching 'M2Crypto/m2crypto.py'
        INFO:add_license_files:adding license file 'LICENCE'
        INFO:execute:writing manifest file 'src/M2Crypto.egg-info/SOURCES.txt'
        INFO:run_command:running build_ext
        INFO:build_extension:building 'M2Crypto._m2crypto' extension
        INFO:swig_sources:swigging src/SWIG/_m2crypto.i to src/SWIG/_m2crypto_wrap.c
        INFO:spawn:swig -python -py3 -I/usr/lib/gcc/x86_64-linux-gnu/12/include -I/usr/local/include -I/usr/include/x86_64-linux-gnu -I/usr/include -D__x86_64__ -I/root/.cache/pypoetry/virtualenvs/donpapi-WjBzr8W3-py3.11/include -I/usr/include/python3.11 -I/usr/include/openssl -includeall -modern -builtin -outdir /tmp/pip-req-build-7ji35ain/src/M2Crypto -o src/SWIG/_m2crypto_wrap.c src/SWIG/_m2crypto.i
        Traceback (most recent call last):
          File "/root/.cache/pypoetry/virtualenvs/donpapi-WjBzr8W3-py3.11/bin/swig", line 5, in <module>
            from swig import swig
        ModuleNotFoundError: No module named 'swig'
        error: command '/root/.cache/pypoetry/virtualenvs/donpapi-WjBzr8W3-py3.11/bin/swig' failed with exit code 1
        [end of output]
    
    note: This error originates from a subprocess, and is likely not a problem with pip.
    ERROR: Failed building wheel for M2Crypto
  Failed to build M2Crypto
  ERROR: Could not build wheels for M2Crypto, which is required to install pyproject.toml-based projects
  

  at /usr/lib/python3/dist-packages/poetry/utils/env.py:1540 in _run
      1536│                 output = subprocess.check_output(
      1537│                     command, stderr=subprocess.STDOUT, env=env, **kwargs
      1538│                 )
      1539│         except CalledProcessError as e:
    → 1540│             raise EnvCommandError(e, input=input_)
      1541│ 
      1542│         return decode(output)
      1543│ 
      1544│     def execute(self, bin: str, *args: str, **kwargs: Any) -> int:

The following error occurred when trying to handle this error:


  PoetryException

  Failed to install /root/.cache/pypoetry/artifacts/de/e9/1c/53d251c759c702dbaaaaca584d20259e0dac1d89660e8cf5208e6d1183/M2Crypto-0.38.0.tar.gz

  at /usr/lib/python3/dist-packages/poetry/utils/pip.py:58 in pip_install
       54│ 
       55│     try:
       56│         return environment.run_pip(*args)
       57│     except EnvCommandError as e:
    →  58│         raise PoetryException(f"Failed to install {path.as_posix()}") from e
       59│ 

  • Installing pyasn (1.6.1)
  • Installing pycryptodome (3.18.0)
  • Installing swig (4.1.1)
  • Updating wheel (0.38.4 -> 0.37.1)
                                    
```

Testing environnement was:

```
git clone https://github.com/login-securite/DonPAPI
cd DonPAPI
poetry update
poetry run DonPAPI
```

**Comment by zblurx** [comment]
Thanks for the report. So I created a branch "dep-fix" with a clean up on dead code and dependancies, could you test if it helps ?

**Comment by zblurx** [comment]
Fixed in c07dd6c01d195ff9bdecc93c5f001dbd63530bae
